kotlinx-coroutines-core/native/src/Exceptions.kt

```diff
@@ -15,6 +15,11 @@ import kotlinx.coroutines.internal.SuppressSupportingThrowableImpl
  */
 public actual typealias CancellationException = kotlin.coroutines.cancellation.CancellationException
 
+@Suppress("INVISIBLE_MEMBER", "INVISIBLE_REFERENCE")
+@kotlin.internal.LowPriorityInOverloadResolution
+public actual fun CancellationException(message: String?, cause: Throwable?): CancellationException =
+    CancellationException(message, cause)
```

Comment:
Is this `LowPriorityInOverloadResolution` just to make this definition work, or is it also so that platform-specific code chooses the other overload?
* If it's for the definition to work, won't a fully-qualified name work better? Like `= kotlin.coroutines.cancellation.CancellationException(message, cause)`.
* If it's for choosing the other overload, maybe we can ensure this also happens on K2 by marking this function as `inline`? Is this possible?

Reply:
To make the declaration work (otherwise, it's conflicting overloads on the declaration level) and, later, the choice of the overload from our/users' code.

>maybe we can ensure this also happens on K2 by marking this function as inline?

It's a bit unclear to me what you've meant, could you please elaborate on that?

Reply:
> It's a bit unclear to me what you've meant, could you please elaborate on that?

If I remember correctly, with K2, in common code, the overload that gets chosen in the end is the common one, so in this case, the `expect` function. If that's undesirable, can't we mark this function as `inline` instead so its calls always get converted to the ones we want?

Reply:
I see, thanks.

>the overload that gets chosen in the end is the common one

Unfortunately, it will be the case only for the code in `kotlinx-coroutines-core` sourceset and not for our users, so I don't think it worth it